doc/services/device_mgmt/smp_groups/smp_group_3.rst

```diff
@@ -0,0 +1,545 @@
+    | ``2``          | Commit settings              |
+    +----------------+------------------------------+
+    | ``3``          | Load/Save settings           |
+    +----------------+------------------------------+
+
+Note that the Zephyr version adds additional commands and features which are not supported by
+the original upstream version, however, the original client functionality should work for
+read/write functionality.
+
+Read/write setting command
+**************************
+
+Read/write setting command allows updating a setting entry on a device or
+getting the current value of a setting from a device.
+
+Read setting request
+====================
+
+Read setting request header fields:
+
+.. table::
+    :align: center
+
+    +--------+--------------+----------------+
+    | ``OP`` | ``Group ID`` | ``Command ID`` |
+    +========+==============+================+
+    | ``0``  | ``3``        | ``0``          |
+    +--------+--------------+----------------+
+
+CBOR data of request:
+
+.. code-block:: none
+
+    {
+        (str)"name"         : (str)
+        (str,opt)"max_size" : (uint)
+    }
+
+where:
+
+.. table::
+    :align: center
+
+    +------------+-----------------------------------------+
+    | "name"     | string of the setting to retrieve       |
+    +------------+-----------------------------------------+
+    | "max_size" | optional maximum size of data to return |
+    +------------+-----------------------------------------+
+
+Read setting response
+=====================
+
+Read setting response header fields:
+
+.. table::
+    :align: center
+
+    +--------+--------------+----------------+
+    | ``OP`` | ``Group ID`` | ``Command ID`` |
+    +========+==============+================+
+    | ``1``  | ``3``        | ``0``          |
+    +--------+--------------+----------------+
+
+CBOR data of successful response:
+
+.. code-block:: none
+
+    {
+        (str)"val"         : (bstr)
+        (str,opt)"limited" : (bool)
+    }
+
+In case of error the CBOR data takes the form:
+
+.. tabs::
+
+   .. group-tab:: SMP version 2
+
+      .. code-block:: none
+
+          {
+              (str)"ret" : {
+                  (str)"group"    : (uint)
+                  (str)"rc"       : (uint)
+              }
+          }
+
+   .. group-tab:: SMP version 1
+
+      .. code-block:: none
+
+          {
+              (str)"rc"       : (int)
+          }
+
+where:
+
+.. table::
+    :align: center
+
+    +------------------+-------------------------------------------------------------------------+
+    | "val"            | binary string of the returned data, note that the underlying data type  |
+    |                  | cannot be specified through this and must be known by the client.       |
+    +------------------+-------------------------------------------------------------------------+
+    | "limited"        | will be set to true if the settings value is larger than the maximum    |
```

Comment:
Because of how the settings subsystem works, it is not necessarily truncated, it just means that the requested maximum data size is more than the Kconfig supported size. I guess it could be changed to truncated if it requested size + 1 and the read size was the size, then it only returned size bytes - thoughts?

Reply:
Ah I understand now. IMHO, because `max_size` is optional, you may just respond with what the real `max_size` was used, in case when it is other than requested.